xds/internal/balancer/edsbalancer/eds_impl.go

```diff
@@ -419,12 +419,24 @@ func (edsImpl *edsBalancerImpl) updateServiceRequestsConfig(serviceName string,
 	if !env.CircuitBreakingSupport {
 		return
 	}
+	edsImpl.pickerMu.Lock()
+	var updatePicker bool
 	if edsImpl.serviceRequestsCounter == nil || edsImpl.serviceRequestsCounter.ServiceName != serviceName {
 		edsImpl.serviceRequestsCounter = client.GetServiceRequestsCounter(serviceName)
+		updatePicker = true
 	}
 	if max != nil {
```

Comment:
Do we need to check `edsImpl.serviceRequestCountMax != *max` before setting `updatePicker = true`?

Reply:
Good catch.

This is also not handling `nil` properly.
Fixed